src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -91,40 +120,35 @@ class PauliTerm:
     def __init__(
         self,
         operator: Union[str, Dict[int, str]],
-        coefficient: complex = 1.0,
+        coefficient: Optional[complex] = None,
     ):
-        self.coefficient = complex(coefficient)
-
-        if isinstance(operator, dict):
-            # Verify key values
-            if not all([qubit_idx >= 0 for qubit_idx in operator]):
-                raise ValueError(
-                    "Invalid qubit index in dictionary keys."
-                    " Make sure all qubit indices are non-negative integers."
-                )
-
-            # Verify value values
-            if not all([op in ALLOWED_OPERATORS for op in operator.values()]):
+        if isinstance(operator, str):
+            _parsed_coefficient, operator = _parse_operators_and_coefficient(operator)
+            if _parsed_coefficient is not None and coefficient is not None:
                 raise ValueError(
```

Comment:
maybe add a test that makes sure this ValueError happens when we pass a coefficient to both the string and argument

Reply:
Suitable test has been added.